public/src/main/java/com/revenuecat/purchases/models/GoogleStoreProduct.kt

```diff
@@ -31,6 +31,9 @@ data class GoogleStoreProduct(
             )
         }
 
+    override val platformProductId: PlatformProductId
+        get() = GooglePlatformProductId(productId)
```

Comment:
How about we just add it as a field and create it when the GoogleStoreProduct/GoogleSubscriptionOption is created: 
e.g. in `ProductDetails.SubscriptionOfferDetails.toSubscriptionOption`